cpp/src/slice2cpp/Gen.cpp

```diff
@@ -256,17 +256,34 @@ namespace
     /// Returns a doxygen formatted link to the provided Slice identifier.
     string cppLinkFormatter(const string& rawLink, const ContainedPtr& source, const SyntaxTreeBasePtr& target)
     {
-        string result = "{@link ";
         if (target)
         {
-            if (auto dataMemberTarget = dynamic_pointer_cast<DataMember>(target))
+            if (dynamic_pointer_cast<DataMember>(target) || dynamic_pointer_cast<Enumerator>(target))
             {
-                // Links to fields must always be qualified in the form 'container#field'.
-                ContainedPtr parent = dynamic_pointer_cast<Contained>(dataMemberTarget->container());
+                ContainedPtr memberTarget = dynamic_pointer_cast<Contained>(target);
+
+                // Links to fields/enumerators must always be qualified in the form 'container#member'.
+                ContainedPtr parent = dynamic_pointer_cast<Contained>(memberTarget->container());
                 assert(parent);
 
                 string parentName = getUnqualified(parent->mappedScoped(), source->mappedScope());
-                return parentName + "#" + dataMemberTarget->mappedName();
+                return parentName + "#" + memberTarget->mappedName();
+            }
+            if (auto enumTarget = dynamic_pointer_cast<Enum>(target))
+            {
+                // If a link to an enum isn't qualified (ie. we're the source and target are in the same module),
+                // we have to place a '#' character in front, to explicitly tell Doxygen it's in the current scope.
+                string link = getUnqualified(enumTarget->mappedScoped(), source->mappedScope());
+                if (link.find("::") == string::npos)
+                {
+                    link.insert(0, "#");
+                }
+                return link;
+            }
```

Comment:
Not sure why we need to do this... But, without this none of the Doxygen links to unqualified enums were working...
And alternative would be to force them to qualified, but I think this is cleaner.